fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/impl/attachment/AttachmentTargetImpl.java

```diff
@@ -16,11 +16,36 @@
 
 package net.fabricmc.fabric.impl.attachment;
 
+import java.util.Map;
+
 import net.minecraft.nbt.NbtCompound;
 
 import net.fabricmc.fabric.api.attachment.v1.AttachmentTarget;
+import net.fabricmc.fabric.api.attachment.v1.AttachmentType;
 
 public interface AttachmentTargetImpl extends AttachmentTarget {
+	/**
+	 * Copies entity attachments when it is respawned and a new instance is created.
+	 * Is triggered on player respawn, return from the End, or entity conversion.
+	 * In the first case, only the attachments with {@link AttachmentType#copyOnPlayerRespawn()} will be transferred.
+	 */
+	@SuppressWarnings("unchecked")
+	static void copyOnRespawn(AttachmentTargetImpl original, AttachmentTargetImpl target, boolean alive) {
+		Map<AttachmentType<?>, ?> attachments = original.fabric_getAttachments();
+
+		for (Map.Entry<AttachmentType<?>, ?> entry : attachments.entrySet()) {
+			AttachmentType<Object> type = (AttachmentType<Object>) entry.getKey();
+
+			if (alive || type.copyOnPlayerRespawn()) {
+				target.setAttached(type, entry.getValue());
+			}
+		}
+	}
```

Comment:
This should only be done for serializable attachments, and it should be done by serializing and deserializing to make sure that the instances remain separate. (For example the data might hold a reference to its target).

This round-trip is why we can only copy serializable attachments in the first place.

Reply:
Wait, what about cases where we always copy (like return from end or mob conversion), wouldn't that mean everything is required to be serializable?